src/Http.ts

```diff
@@ -17,28 +21,15 @@ export default class Http {
     })
   }
 
-  async get(q, params = {}) {
+  async spreeResponse() {
```

Comment:
I think there's a chance of `this.method`, `this.route` and `this.params` being replaced when making multiple sync `spreeResponse()` calls. I suggest sending these values as parameters to `spreeResponse`. Also, they should all be cleared after the request is made (not an issue if `spreeResponse` received parameters instead of using state).

Reply:
@tniezg but this `class method`, not a `function`. I really don't like mixed two styles. If use `class` we should use an instance variable, if we use functions we pass parameters

Reply:
@igorbezsmertnyi 

"If use `class` we should use an instance variable, if we use functions we pass parameters" - I wholeheartedly disagree with this statement. Especially here.

It looks like this code will work after all, I misjudged at first. I'd rather not put transient values, such as `this.params`, in state, though. They're only required to make the call to Axios. All the state values used by `spreeResponse` need to be reset before it's called, because not all callers replace all these values. For example, `removeStoreCredits ` doesn't replace `this.params`.